versions/schematic-2.md

```diff
@@ -0,0 +1,196 @@
+<a name="schematicBiomeData"></a>BiomeData | `varint[]` | Specifies the main storage array which contains `Width * Height * Length` entries for `Biome`s at positions. Each entriy is specified as a varint and refers to an index within [BiomePalette](#schematicBiomePalette). The entries are indexed by `x + z * Width + y * Width * Length`.
+#### <a name="metadataObject"></a> Metadata Object
+
+An object which provides optional additional meta information about the schematic. The fields outlined here are guidelines to assist with standardization but it is recommended that any program reading and writing schematics persist all fields found within this object.
+
+##### Fields
+
+Field Name | Type | Description
+---|:---:|---
+<a name="metadataName"></a>Name | `string` | The name of the schematic.
+<a name="metadataAuthor"></a>Author | `string` | The name of the author of the schematic.
+<a name="metadataDate"></a>Date | `long` | The date that this schematic was created on. This is specified as milliseconds since the Unix epoch.
+<a name="metadataRequiredMods"></a>RequiredMods | `string`[] | An array of mod ids which have blocks which are referenced by this schematic's defined [Palette](#schematicPalette).
+
+##### Metadata Object Example:
+
+```js
+{
+    "Name": "My Schematic",
+    "Author": "Author Name",
+    "RequiredMods": [
+        "a_mod",
+        "another_mod"
+    ]
+}
+```
+
+#### <a name="paletteObject"></a>Palette Object
+
+An object which holds a mapping of a block state id to an index. The indices are recommended to start at `0` and may be no more than [`PaletteMax - 1`](#schematicPaletteMax). If the palette is not specified then the global mapping of ids is used instead. This is not recommended as it both expands the size of your blockdata and removes cross server and cross version compatibility for modded blocks which may change ids due to a variety of reasons.
+
+#### Block State Ids
+
+The format of the Block State identifier is the id of the block type and a set of comma-separated property `key=value` pairs surrounded by square brackets. If the block has no properties then they can be excluded. The block type id is specified as a [Resource Location](#defResourceLocation).
+
+For example the air block has no properties so its id representation would be just the block type id `minecraft:air`. The planks block however has an enum property for the `variant` so its id would be `minecraft:planks[variant=oak]`. Properties should be ordered with their keys in alphabetical ordering.
```

Comment:
This example doesn't work well in past 1.12 (planks no longer have a variant property), though I'm not sure how much that matters.